datafusion/physical-plan/src/limit.rs

```diff
@@ -442,7 +442,7 @@ impl LimitStream {
 
             match &poll {
                 Poll::Ready(Some(Ok(batch))) => {
-                    if batch.num_rows() > 0 && self.skip == 0 {
```

Comment:
Is the rationale that this condition is redundant because any batch with zero rows will also have self.skip equal to zero?

Reply:
Yes. To my understanding these two conditions are identical. So I removed one to make it simpler